attestation-service/src/token/mod.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::*;
+use serde::Deserialize;
+use serde_json::Value;
+use strum_macros::EnumString;
+
+mod simple;
+
+const DEFAULT_TOKEN_TIMEOUT: i64 = 5;
+
+pub trait AttestationTokenBroker {
+    /// Issue an signed attestation token with custom claims.
+    /// Return base64 encoded Json Web Token.
+    fn issue(&self, custom_claims: Value) -> Result<String>;
+
+    /// Get the public keys and X.509 formatted certificate chain of the attestation token broker.
+    /// Returns the certificate chain in [JWKS format](https://www.rfc-editor.org/rfc/rfc7517#appendix-B).
+    fn pubkey_jwks(&self) -> Result<String>;
+}
+
+#[derive(Deserialize, Debug, Clone, EnumString)]
+pub enum AttestationTokenBrokerType {
+    Simple,
```

Comment:
What other types do you anticipate? Are we going to be proxying this trait to some other service at some point?

Reply:
Currently, we use a simple random RS384 key signature, but in real deployment scenarios, there may be more demands for the authority of AS.

For example, in a typical deployment scenario, KBS owner hands over AS to another owner he trust for deployment, and KBS owner needs to confirm that the verification result of TEE evidence must come from the AS he trust. In this case, there are stricter requirements for the content and signature of the token, and a new Token Broker trait object may need to be implemented to sign the token using the AS owner's private key, and attach detailed owner identity information and a trusted CA guaranteed public key certificate chain in the token.

The above is just an example. In practical productization, there may be flexibility. It is possible that the issuer of the token is an independent service center in the AS deployment environment, which is the case of remote connection you mentioned.